src/storage/adapters/gridfs.ts

```diff
@@ -1,17 +1,57 @@
-class GridFS implements StorageAdapter {
+import { ReadStream, WriteStream } from 'fs';
+import StorageAdapter from './interface';
+import {
+    Db,
+    ObjectId,
+    MongoClient,
+    GridFSBucket,
+    GridFSBucketReadStream,
+    GridFSBucketWriteStream
+} from 'mongodb'
+
+export default class GridFS implements StorageAdapter {
+    private client: MongoClient;
+    private ready: Promise<void>;
+    private db!: Db;
+    private bucket!: GridFSBucket;
     constructor() {
-        throw new Error("Method not implemented.");
+        if (process.env.GRIDFS_PATH === undefined) throw Error("GRIDFS_PATH is not a defined env variable!");
+        this.client = new MongoClient(process.env.GRIDFS_PATH);
+        this.ready = new Promise(resolve => {
+            this.client.connect(() => {
+                this.db = this.client.db("file_storage");
+                this.bucket = new GridFSBucket(this.db);
+                resolve();
+            });
+        });
     }
-    async add(id: string): Promise<string> {
-        throw new Error("Method not implemented.");
+
+    async add(fileStream: ReadStream): Promise<string> {
+        await this.ready;
+        const stream: GridFSBucketWriteStream = this.bucket.openUploadStream("untitled");
+        return new Promise((resolve, reject) => {
+            fileStream.pipe(stream).on('finish', () => {
+                resolve(stream.id.toString());
+            }).on('error', (error) => {
+                reject(error);
+            });
+        })
     }
-    async get(id: string): Promise<StorageFile> {
-        throw new Error("Method not implemented.");
-    }
-    async list(id: string): Promise<StorageFile[]> {
-        throw new Error("Method not implemented.");
+
+    async get(id: string, stream: WriteStream): Promise<void> {
```

Comment:
Wouldnt it be better if we returned the stream itself (or a promised version of it)?

Reply:
Couldn't get it to work.